Makefile

```diff
@@ -109,27 +109,25 @@ k8s: validate ## Build default K8s version of EKS Optimized AL2 AMI
 	@echo "$(T_GREEN)Building AMI for version $(T_YELLOW)$(kubernetes_version)$(T_GREEN) on $(T_YELLOW)$(arch)$(T_RESET)"
 	$(PACKER_BINARY) build -timestamp-ui -color=false $(PACKER_VAR_FLAGS) $(PACKER_TEMPLATE_FILE)
 
-# Build dates and versions taken from https://docs.aws.amazon.com/eks/latest/userguide/install-kubectl.html
-
 .PHONY: 1.23
 1.23: ## Build EKS Optimized AL2 AMI - K8s 1.23
-	$(MAKE) k8s kubernetes_version=1.23.17 kubernetes_build_date=2023-08-16
+	$(MAKE) k8s $(shell hack/latest-binaries.sh 1.23)
```

Comment:
since these all pretty much look the same is there anything stopping us from generating these targets using the [`kubernetes-versions.json`](https://github.com/awslabs/amazon-eks-ami/blob/master/kubernetes-versions.json)? i think it might look like

```Make
# generate k8s version targets from versions json
K8S_VERSIONS=$(shell cat kubernetes-versions.json | jq -r -c 'join(" ")')
.PHONY: $(K8S_VERSIONS)
$(K8S_VERSIONS): %:
	$(MAKE) k8s $(shell hack/latest-binaries.sh $@)
```

and once this is rebased we'd need to add `1.28` to the `kubernetes-version.json`!

Reply:
We could, yeah; but that JSON file controls which versions are used for CI, which in turn depends on `eksctl` supporting the version, so we can't couple the make targets with that list.

We could avoid a dependency on the JSON list if we go with a make variable instead of a target, i.e.:
```
make k8s=1.27
```

instead of:
```
make 1.27
```

Reply:
gotcha, so is there any validation required around that input? or is it just based on whether version/binaries can be found for a given `k8s=*`, and if so then it can be built

Reply:
Yep, we can just bail if there are no binaries available. I added this in the latest rev 👍